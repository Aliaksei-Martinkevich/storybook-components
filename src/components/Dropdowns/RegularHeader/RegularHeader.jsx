import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './RegularHeader.scss';


const RegularHeader = ({ text, className = '', expanded = false }) => (
  <div className={classNames('dropdown-regular-header', className, {
    expanded,
    collapsed: !expanded,
  })}
  >
    {text}
  </div>
  );

RegularHeader.propTypes = {
  text: PropTypes.string,
  className: PropTypes.string,
  expanded: PropTypes.bool,
};

export default RegularHeader;
