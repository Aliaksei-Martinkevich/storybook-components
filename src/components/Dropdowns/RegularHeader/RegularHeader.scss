.dropdown-regular-header {
    box-sizing: border-box;
    color: #BBBBBB;
    border: 1px solid #CCCCCC;
    padding: 5px;
    &.collapsed {
        background-image: url('../../../../assets/icons/ico-expand-arrow.png');
        background-repeat: no-repeat;
        background-position: calc(100% - 5px) center;
        background-size: 15px;
    }
    &.expanded {
        background-image: url('../../../../assets/icons/ico-collapse-arrow.png');
        background-repeat: no-repeat;
        background-position: calc(100% - 5px) center;
        background-size: 15px;
    }
}