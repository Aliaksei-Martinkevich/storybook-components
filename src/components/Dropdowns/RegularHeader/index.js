import React from 'react';
import PropTypes from 'prop-types';

import './index.css';


const RegularHeader = ({ text, className = ''}) => {
  return (
    <div className={`dropdown-regular-header ${className}`}>
      {text}
    </div>
  )
};

RegularHeader.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.string,
  className: PropTypes.string,
  remark: PropTypes.string,
};

export default RegularHeader;
