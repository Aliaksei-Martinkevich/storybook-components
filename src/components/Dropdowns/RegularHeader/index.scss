.dropdown-regular-header {
    color: #BBBBBB;
    border: 1px solid #CCCCCC;
    padding: 5px;
    display: inline-block;
    min-width: 340px;
}

.collapsed .dropdown-regular-header {
    background-image: url('../../../../public/ico-expand-arrow.png');
    background-repeat: no-repeat;
    background-position: calc(100% - 5px) center;
    background-size: 15px;
}

.expanded .dropdown-regular-header {
    background-image: url('../../../../public/ico-collapse-arrow.png');
    background-repeat: no-repeat;
    background-position: calc(100% - 5px) center;
    background-size: 15px;
}