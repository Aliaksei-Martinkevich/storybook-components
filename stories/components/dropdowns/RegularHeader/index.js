import React from 'react';
import { storiesOf } from '@storybook/react';

import Components from '../../../../src/components'
const RegularHeader = Components.Dropdowns.RegularHeader;

export default storiesOf('Dropdown Regular Header', module)
  .add('Expanded', () => <div className='expanded'><RegularHeader text='Expanded header' /></div>)
  .add('Collapsed', () => <div className='collapsed'><RegularHeader text='Collapsed header' /></div>);