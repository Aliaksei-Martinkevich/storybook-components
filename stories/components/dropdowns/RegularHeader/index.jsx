import React from 'react';
import { storiesOf } from '@storybook/react';

import RegularHeader from '../../../../src/components/Dropdowns/RegularHeader/RegularHeader.jsx';

export default storiesOf('Dropdown Regular Header', module)
  .add('Expanded', () => <div className="expanded"><RegularHeader text="Expanded header" /></div>)
  .add('Collapsed', () => <div className="collapsed"><RegularHeader text="Collapsed header" /></div>);
