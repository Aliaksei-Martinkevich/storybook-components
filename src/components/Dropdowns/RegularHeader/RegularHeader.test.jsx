import React from 'react';
import renderer from 'react-test-renderer';
import RegularHeader from './RegularHeader.jsx';

describe('Regular header', () => {
  it('Should be collapsed', () => {
    const customClassName = 'class-name-example';
    const text = 'Some text';
    const component = renderer.create(
      <RegularHeader
        className={customClassName}
        text={text}
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should be expanded', () => {
    const customClassName = 'class-name-example';
    const text = 'Some text';
    const component = renderer.create(
      <RegularHeader
        className={customClassName}
        text={text}
        expanded
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});