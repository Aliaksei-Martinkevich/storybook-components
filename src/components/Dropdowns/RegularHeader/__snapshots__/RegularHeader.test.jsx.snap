// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regular header Should be collapsed 1`] = `
<div
  className="dropdown-regular-header class-name-example collapsed"
>
  Some text
</div>
`;

exports[`Regular header Should be expanded 1`] = `
<div
  className="dropdown-regular-header class-name-example expanded"
>
  Some text
</div>
`;
